agent/connect/ca/testing.go

```diff
@@ -194,13 +194,13 @@ func (v *TestVaultServer) WaitUntilReady(t testing.T) {
 	retry.Run(t, func(r *retry.R) {
 		resp, err := v.client.Sys().Health()
 		if err != nil {
-			r.Fatalf("err: %v", err)
+			r.Errorf("err: %v", err)
 		}
 		if !resp.Initialized {
-			r.Fatalf("vault server is not initialized")
+			r.Error("vault server is not initialized")
 		}
 		if resp.Sealed {
-			r.Fatalf("vault server is sealed")
+			r.Error("vault server is sealed")
```

Comment:
I noticed this test flaking often and believe the retry is never run because `Fatalf` panics and kills the loop

Reply:
> believe the retry is never run because Fatalf panics and kills the loop

If we were calling `t.Fatalf` then that would be the case. `retry.Run` however will recover from the panic induced by the `r.Fatalf` call and continue retrying until it thinks it should stop (after 7s in this case)

Reply:
Ah I see! Dropped this commit